### PR TITLE
Run all libzfs things in a single thread

### PR DESCRIPTION
**Pull request by william-gr**
Ticket: #26774

### CODE REVIEW

File: src/middlewared/middlewared/main.py [hunk 1]
```diff
@@ -859,32 +862,41 @@ def get_schema(self, name):
             if asyncio.iscoroutinefunction(methodobj):
                 return await methodobj(*args)
             else:
-                return await self.threaded(methodobj, *args)
+                pool = None
+                if hasattr(methodobj, '_threaded'):
+                    pool = methodobj._threaded
+                if serviceobj._config.thread_pool:
+                    pool = serviceobj._config.thread_pool
```
Comment (themylogin) [inline src/middlewared/middlewared/main.py]: Maybe method-specific pool should have more priority over service-specific as it is more local?

File: src/middlewared/middlewared/main.py [hunk 2]
```diff
@@ -859,32 +862,41 @@ def get_schema(self, name):
             if asyncio.iscoroutinefunction(methodobj):
                 return await methodobj(*args)
             else:
-                return await self.threaded(methodobj, *args)
+                pool = None
+                if hasattr(methodobj, '_threaded'):
+                    pool = methodobj._threaded
```
Comment (themylogin) [inline src/middlewared/middlewared/main.py]: Confusing property name `_threaded` that stores thread pool

File: src/middlewared/middlewared/main.py [hunk 3]
```diff
@@ -823,18 +823,21 @@ def add_schema(self, schema):
     def get_schema(self, name):
         return self.__schemas.get(name)
 
-    async def threaded(self, method, *args, **kwargs):
+    async def _threaded(self, pool, method, *args, **kwargs):
```
Comment (themylogin) [inline src/middlewared/middlewared/main.py]: How about renaming these to `self.run_in_thread` and `self.run_in_thread_pool`?

Comment (william-gr) [reply src/middlewared/middlewared/main.py]: `threaded` is already in use by many things, I did not want to change that.

File: src/middlewared/middlewared/main.py (hunk 1 above)
Comment (william-gr) [reply src/middlewared/middlewared/main.py]: Agreed!

File: src/middlewared/middlewared/main.py (hunk 2 above)
Comment (william-gr) [reply src/middlewared/middlewared/main.py]: True!

**Comment by william-gr** [comment]
@themylogin does that address most of your concerns? I did not want to change `threaded` method name because as explained but you can convince me otherwise ;)

**Comment by themylogin** [comment]
@william-gr it's not an ambiguous name and can be renamed automatically. Verbs (in my opinion) are usually better method names than adjectives. Both `threaded` and `_threaded` can be useful, there is no need to hide second one and it's name does not tell us how it is different from `_threaded`.

**Comment by william-gr** [comment]
@themylogin Ok, you convinced me ;) How does that look now?

**Comment by themylogin** [comment]
@william-gr looks great now, thank you!

**Comment by william-gr** [comment]
Jenkins retest this please

**Pull request merged**